segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookKeeperLog.java

```diff
@@ -396,13 +473,17 @@ private void handleWriteException(int responseCode, Write write) {
         try {
             if (ex instanceof BKException.BKLedgerFencedException) {
                 // We were fenced out.
-                ex = new DataLogWriterNotPrimaryException("Log not primary anymore.", ex);
+                ex = new DataLogWriterNotPrimaryException("BookKeeperLog is not primary anymore.", ex);
```

Comment:
As explained above, fencing triggers a separate exception which causes a container shutdown. I don't think this will be an issue.